src/customJs/properties/promo_codes/types.ts

```diff
@@ -1,8 +1,18 @@
-import { EMPTY_SELECTION } from '../../constants';
+import { EMPTY_SELECTION, PROMO_CODE_TYPE } from '../../constants';
 
 export type PromoCodesValue = string | EMPTY_SELECTION;
 export type StoresValue = string | EMPTY_SELECTION;
 
 export type StoreDependentToolValues = {
   store: StoresValue;
 };
+
+export type PromoCodeItem = {
+  code: string;
+  type: PROMO_CODE_TYPE;
```

Comment:
I would keep all the type definitions together here. I would revert the changes in constants.ts

```suggestion
  type: 'percent' | 'money;
```

Reply:
Done.